src/diffusers/pipelines/blip_diffusion/pipeline_blip_diffusion.py

```
@@ -0,0 +1,337 @@
+from ...utils import (
+    BaseOutput,
+    is_accelerate_available,
+    is_accelerate_version,
+    logging,
+    randn_tensor,
+    replace_example_docstring,
+)
+from ...utils.pil_utils import PIL_INTERPOLATION
+from torch import nn
+from transformers.activations import QuickGELUActivation as QuickGELU
+from .modeling_blip2 import Blip2QFormerModel
+import tqdm
+from torchvision import transforms
+from torchvision.transforms.functional import InterpolationMode
+from ...image_processor import BlipImageProcessor
+from PIL import Image
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+import re
+
+def prepare_cond_image(
+        image, width, height, batch_size, device, do_classifier_free_guidance=True
+    ):
+        if not isinstance(image, torch.Tensor):
+            if isinstance(image, Image.Image):
+                image = [image]
+
+            if isinstance(image[0], Image.Image):
+                images = []
+
+                for image_ in image:
+                    image_ = image_.convert("RGB")
+                    image_ = image_.resize(
+                        (width, height), resample=PIL_INTERPOLATION["lanczos"]
+                    )
+                    image_ = np.array(image_)
+                    image_ = image_[None, :]
+                    images.append(image_)
+
+                image = images
+
+                image = np.concatenate(image, axis=0)
+                image = np.array(image).astype(np.float32) / 255.0
+                image = image.transpose(0, 3, 1, 2)
+                image = torch.from_numpy(image)
+            elif isinstance(image[0], torch.Tensor):
+                image = torch.cat(image, dim=0)
+
+        image_batch_size = image.shape[0]
+
+        if image_batch_size == 1:
+            repeat_by = batch_size
+        else:
+            # image batch size is the same as prompt batch size
+            # repeat_by = num_images_per_prompt
+            raise NotImplementedError
```

Maybe raise this error with a better error message?